### ISSUE TITLE
python2 requirements not in setup.py

### ISSUE THREAD

**Issue by jcrowgey** [andrivet/python-asn1#22]
I rely on this package for a project which aims to maintain compatibility with py2 and py3, so I list 'asn1' in my install_requires.  When installing in python3, that works perfectly!  When installing in python2, I'm missing the 'futures' dependency which you guys (helpfully) list on the README here.

Would you guys accept a pull request against your setup.py which adds futures when installing in python2?

**Comment by andrivet** [comment]
Futures is listed in requirements.txt.

**Comment by jcrowgey** [comment]
Indeed.  But requirements.txt is not referenced in your setup.py.  So if I pip install your package in a python2 enironment, it is not runnable:

```
$ virtualenv venv
Running virtualenv with interpreter /usr/bin/python2
...
Installing setuptools, pkg_resources, pip, wheel...done.
$ . venv/bin/activate
(venv) $ pip install asn1
Collecting asn1
  Downloading https://files.pythonhosted.org/packages/98/90/1fef44666b33b3127cc25d32a8b0ded19676c70a6980b5be9cea27470a26/asn1-2.2.0-py2.py3-none-any.whl
Installing collected packages: asn1
Successfully installed asn1-2.2.0
(venv) $ python -c 'import asn1'
Traceback (most recent call last):
...
ImportError: No module named builtins
```

As we know, this can be rectified by installing futures in the virtualenv, but ideally, your setup.py should list what is required to run the code so that pip install works out of the box.

I may be mistaken, but I think it's traditional to list development requirements in requirements.txt (so that if I git clone this repo I know what to install in order to hack on this project), but setup.py contains the list of packages required at run time, so that pip knows what to pull down when installing on an end-user's environment.

I'd love to help!

**Comment by jcrowgey** [comment]
I no longer work on the project I was writing about in 2018, I can't tell you if they still require Python2.
